//
// This file defines the configuration format for initializing the
// plugin.
//

import "google/protobuf/descriptor.proto";

message DeviceDescriptor {
    required uint32 vendor_id = 1;
    required uint32 product_id = 2;
}

message Configuration {
    // regexes of allowed/forbidden urls
    repeated bytes whitelist_urls = 1;
    repeated bytes blacklist_urls = 2;

    // compiled specification of the wire protocol, serialized
    // FileDescriptorSet can be generated with `protoc -o`
    required google.protobuf.FileDescriptorSet wire_protocol = 3;

    // descriptors of allowed devices
    repeated DeviceDescriptor known_devices = 4;

    // timestamp of expiration
    optional int32 valid_until = 5;
}